src/frequenz/sdk/actor/power_distributing/power_distributing.py

```diff
@@ -132,8 +132,32 @@ async def _run(self) -> None:  # pylint: disable=too-many-locals
         """
         await self._component_manager.start()
 
+        power_tasks: dict[str, asyncio.Task[None]] = {}
+
+        def cleanup(task: asyncio.Task[None]) -> None:
+            task_name = task.get_name()
+            assert task_name in power_tasks, "Task not found in power_tasks"
+
+            # Raise if there was an exception while processing the task
+            task.result()
+
+            del power_tasks[task_name]
+
         async for request in self._requests_receiver:
-            await self._component_manager.distribute_power(request)
+            task_name = f"power_request_ids={request.component_ids}"
+            if task_name in power_tasks:
+                result = Error(
+                    request=request,
+                    msg="A request for the same components is being processed. Skipped.",
+                )
+                await self._result_sender.send(result)
+                continue
```

Comment:
Wouldn't it be better to cancel the ongoing request and do the new one instead? Maybe not because we can end up with no request being finished if we have a burst of requests?

Reply:
I agree that the latest one needs to be executed, but instead of cancelling the old one, I think we should keep the latest request and send it as soon as the running one is complete.  That is, discard only the older requests and execute the latest.

Reply:
> Maybe not because we can end up with no request being finished if we have a burst of requests?
Indeed, my thought here was to avoid entering a loop of cancelling the in-progress request when a new request arrives. This is a bit tricky because we don't really know the state of the current processing request so I decided to ignore/skip the new request. This is the point we all need to agree on.  @sahas-subramanian-frequenz any thoughts on this?

> Wouldn't it be better to cancel the ongoing request and do the new one instead?
According to the outdated documentation, the actor was doing this. And the current state is the actor is just queuing up requests (without this patch)

Reply:
@shsms I haven't seen your comment when I replied to this thread

Reply:
Absolutely, I agree with the reasoning above on why cancelling doesn't make sense.  We shouldn't cancel.

Reply:
Updated to keep track of the latest pending power request